File: solutions/kattis/naq14/8queens_lok.c
#include<stdio.h>

void main()
{
    int pos[64][2];
    short invalid = 0;
    int i;
    int queen_count = 0;
    for (i = 0;i < 8;i++)
    {
        char line[8];
        int j;
        scanf("%s", &line);
        
        for (j = 0;j < 8; j++)
        {
            if (line[j] == '*')
            {
                pos[queen_count][0] = i;
                pos[queen_count][1] = j;
                queen_count++;
            }
        }
    }
        
    if (queen_count != 8) invalid = 1;
    
    i = 0;
    for(i = 0;i < 8 && !invalid;i++)
    {
        int j;
        for (j = i+1;j < 8 && !invalid;j++) invalid = pos[i][0] == pos[j][0] || pos[i][1] == pos[j][1] || pos[i][0]-pos[j][0] == pos[i][1]-pos[j][1] || pos[i][0]+pos[i][1] == pos[j][0]+pos[j][1];
    }

    if (invalid) printf("invalid");
    else printf("valid");
}

File: solutions/kattis/naq14/8queens_lok.py
pos = []
invalid = 0
for i in range(0,8):
    line = input()
    line = list(line)
    count = 0
    for j in range(0, 8):
        if line[j] == '*':
            pos.append([i,j])
if len(pos) != 8:
    invalid = 1
i = 0
while i < len(pos) and not invalid:
    j = i + 1
    while j < len(pos) and not invalid:
        invalid = pos[i][0] == pos[j][0] or pos[i][1] == pos[j][1] or pos[i][0]-pos[j][0] == pos[i][1]-pos[j][1] or pos[i][0]+pos[i][1] == pos[j][0]+pos[j][1]
        j += 1
    i += 1

if invalid:
    print("invalid")
else:
    print("valid")